htdocs/loan/class/loanschedule.class.php

```diff
@@ -305,7 +305,7 @@ public function update($user = 0, $notrigger = 0)
 			$this->amount_interest = trim($this->amount_interest);
 		}
 		if (isset($this->num_payment)) {
-			$this->num_payment = trim($this->num_payment);
+			$this->num_payment = (int) $this->num_payment;
```

Comment:
I think num_payment can be ABC123, so string

Reply:
Correct. This is indeed something we already saw in the past: https://github.com/Dolibarr/dolibarr/commit/aafe309b5814de27a6aced88a507747463b25878.

Now corrected.